### ISSUE TITLE
[Suggestion]: Player Size Scaling

### ISSUE THREAD

**Issue by JonasL0430** [Yesssssman/epicfightmod#1654]
### Have you checked if there's any similar suggestions ?

- [X] I checked, there are no similar suggestions that have been reported.

### Are you using the latest Epic Fight ?

- [X] I checked I'm using latest Epic Fight version.

### Is this suggestion related to another mod? (i.e: add animations for X mod)

- [X] This is related to other mods.

### The mod

https://www.curseforge.com/minecraft/mc-mods/pehkui

### Minecraft Version

1.20.1

### What's your suggestion?

Pehkui is a mod that allows the size of the player to be changed. I’ve seen a few issues posted about pehkui compat, but surprisingly I haven’t seen one about the most important issue: colliders. Pehkui almost works perfectly, it’s so close, it’s just that the attack colliders don’t scale with the player causing it to be impossible to attack, especially when at large scales. This is because the the colliders still retain their original size and only move as if the player was swinging the weapon at the original size. Perhaps it's also worth noting that the colliders appear to be in the right place when turning hitboxes on using f3+B. These boxes follow the resized player attacks perfectly, but don't do any damage as the "actual" colliders still seem to be at the player's original size.

However, it may be easier for epic fight to simply have its own option to scale the player. It already uses a custom model for the player, and it’s already very easy to scale basically every other entity using datapacks and the “scale” attribute. So it would be really nice if, for example, a “scale” attribute could be added which could be applied to any entity (including the player) via in game commands. Just like the other attributes like weight penalty and so on.

**Comment by guivnf** [comment]
duplicate of #1303

**Comment by JonasL0430** [comment]
So I noticed that issue 1303 is supposed to be fixed in the next update. With that said, that issue seems to be about first person view, not colliders. Additionally, I've tested this issue that I posted in the early access version of the next update available through patreon, and it seems to still not work. Is there (or will there be in the foreseeable future) any way to scale up the player properly? While retaining collider and hit box functionality?

**Comment by guivnf** [comment]
> So I noticed that issue 1303 is supposed to be fixed in the next update. With that said, that issue seems to be about first person view, not colliders. Additionally, I've tested this issue that I posted in the early access version of the next update available through patreon, and it seems to still not work. Is there (or will there be in the foreseeable future) any way to scale up the player properly? While retaining collider and hit box functionality?

The fix that was mentioned is not yet available for the public. It's retained in our development enviroment and we'll release it once it's stable enough for use. Though it's marked as solved you can't really access it yet. The complete feature list is stated in the patreon post.

**Comment by JonasL0430** [comment]
Thank you for the information. Just to clarify, I have heard about the first person view being overhauled in an upcoming update. 

However, are you also saying that full Pehkui mod compatibility (colliders and first person view) is coming in a later update?

To know that pehkui compat has been fixed in a dev environment would be huge for planning my current project. 

I did read through the changes listed on patreon and I didn't see anything related to pehkui. There's the mermaid mod compat changes which sound kinda related, but from my testing pehkui compat is still broken with the latest early access version of epic fight.
